src/model/genesis/AGenesis.java

```diff
@@ -277,4 +278,38 @@ public int maxGeneration(){
 		return this.maxGen;
 	}
 
+	private void meetPeople(Random r) {
+		// efficiency... I got nothing. so this is a stopgap until we can talk about it some.
+		List<IPerson> people = livingPopulation();
+
+		for (int attempt = 0; attempt < MEETING_ATTEMPTS_PER_YEAR; attempt++) {
+			int max = people.size()/2;
+			Collections.shuffle(people);
+			for (int i = 0; i < max; i++) {
+				IPerson p1 = people.get(i);
+				IPerson p2 = people.get(i * 2);
+				double chance = IRelationship.meetingChance(p1, p2);
+				double roll = r.nextDouble();
+				if (roll < chance) {
+					p1.meet(p2, this.timeInYears);
+				}
+			}
+		}
+	}
+
+	private void progressRelationships(Random r) {
+		// efficiency considerations? because this is gonna somehow have to go through every relationship in the pop
+		// or maybe for every person, it picks a max number of relationships to change?
+		// so like go through each persons' list of relationships, and for the ones that are chosen, progress relationship?
+
+		for (IPerson p : this.livingPopulation()) {
+			int numRelationships = Math.min(p.getRelationships().size(), RELATIONSHIP_CHANGES_PER_YEAR);
+			Iterator<IRelationship> it = p.getRelationships().values().iterator();
```

Comment:
I believe values() is iterable so you should be able to just use a foreach loop.

Reply:
I don't necessarily want to go through all of the relationships, though, I only want to go as high as the `RELATIONSHIP_CHANGES_PER_YEAR`. Would it be better to break out of a foreach loop when we get to more than that, or is there a better way to do that?